### PR TITLE
improve vertical positioning & paren shaping in `syntax->string`

### PR DESCRIPTION
**Pull request by mbutterick**
Two improvements:

1) Currently, `syntax->string` collapses multiple newlines into one. But the docs say that it “Builds a string with newlines and indenting according to the source locations”. So it seems more accurate to preserve the newlines.

2) Currently, `syntax->string` ignores the `'paren-shape` property. (Perhaps it predates 'paren-shape.) It seems more accurate to observe it.

### CODE REVIEW

File: racket/collects/syntax/to-string.rkt [hunk 1]
```diff
@@ -55,10 +55,14 @@
                  ((loop init-line!) i))]
               [(pair? (syntax-e c))
                (advance c init-line!)
-               (printf "(")
+               (define c-paren-shape (syntax-property c 'paren-shape))
+               (printf (format "~a" (or c-paren-shape #\()))
```
Comment (AlexKnauth) [inline racket/collects/syntax/to-string.rkt]: `printf` already handles format strings, so this could be `(printf "~a" (or c-paren-shape #\( ))`.

**Comment by mflatt** [comment]
I didn't know that this function exists.

@stamourv, "optimization-coach" has the only use of syntax->string` that I find. Ok with you?

**Comment by stamourv** [comment]
Yes.

**Comment by AlexKnauth** [comment]
What happened to this?

**Comment by stamourv** [comment]
Merged, thanks!

**Comment by AlexKnauth** [comment]
As https://github.com/racket/racket/commit/9422d6660103494e6c79f9f3e1c8944de66c4901

**Pull request closed without merging**